app/models/user.rb

```diff
@@ -135,6 +135,10 @@ def resubmit!(organization = nil)
     assignments.each { |it| it.notify! rescue nil }
   end
 
+  def currently_in_exam?
+    Exam.any? { |e| e.in_progress_for? self }
```

Comment:
Also, to avoid making this query all the time (as the navigation bar is displayed always), we could only check it on the discussions controller and display a more descriptive screen (instead of forbidden) when they try to access it.